Create the resource Address groups

diogoEsteves42 opened the pull request:
<!-- Put a description of what this PR is for here, along with any references to issues that this resolves or contributes to -->
Adding network security `AddressGroups` resource support.

fixes: https://github.com/hashicorp/terraform-provider-google/issues/14299



<!--
Replace each [ ] with [X] to check it. Switch to the preview view to make it easier to click on links.
These steps will speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.
-->
If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none

Unless you choose release-note:none, please add a release note.

See .ci/RELEASE_NOTES_GUIDE.md for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_network_security_address_group`
```

modular-magician commented:
Hello!  I am a robot who works on Magic Modules PRs.

I've detected that you're a community contributor. @SarahFrench, a repository maintainer, has been assigned to assist you and help review your changes. 

<details>
  <summary>:question: First time contributing? Click here for more details</summary>

---

Your assigned reviewer will help review your code by: 
* Ensuring it's backwards compatible, covers common error cases, etc.
* Summarizing the change into a user-facing changelog note.
* Passes tests, either our "VCR" suite, a set of presubmit tests, or with manual test runs.

You can help make sure that review is quick by running local tests and ensuring they're passing in between each push you make to your PR's branch. Also, try to leave a comment with each push you make, as pushes generally don't generate emails.

If your reviewer doesn't get back to you within a week after your most recent change, please feel free to leave a comment on the issue asking them to take a look! In the absence of a dedicated review dashboard most maintainers manage their pending reviews through email, and those will sometimes get lost in their inbox.

---

</details>

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7713-old..auto-pr-7713) ( 2 files changed, 159 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7713-old..auto-pr-7713) ( 6 files changed, 1008 insertions(+), 2 deletions(-))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-7713-old..auto-pr-7713) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-7713-old..auto-pr-7713) ( 8 files changed, 219 insertions(+))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_network_security_address_group` (4 total tests)
Untested fields: `self_link`, `labels`

Please add acceptance tests which include these fields.

modular-magician commented:
#### Tests analytics 
Total tests: `2623` 
Passed tests `2324` 
Skipped tests: `277` 
Affected tests: `22` 

#### Action taken 
<details> <summary>Found 22 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccDataSourceGoogleFirebaseAndroidAppConfig|TestAccNetworkSecurityAddressGroups_update|TestAccNetworkSecurityAddressGroup_networkSecurityAddressGroupsAdvancedExample|TestAccNetworkSecurityAddressGroup_networkSecurityAddressGroupsBasicExample|TestAccSqlDatabaseInstance_activationPolicy|TestAccSqlDatabaseInstance_rootPasswordShouldBeUpdatable|TestAccSqlDatabaseInstance_updateReadReplicaWithBinaryLogEnabled|TestAccSqlDatabaseInstance_Timezone|TestAccSqlDatabaseInstance_PointInTimeRecoveryEnabledForSqlServer|TestAccSqlDatabaseInstance_PointInTimeRecoveryEnabled|TestAccSqlDatabaseInstance_withPrivateNetwork_withoutAllocatedIpRange|TestAccSqlDatabaseInstance_authNets|TestAccSqlDatabaseInstance_settingsDowngrade|TestAccSqlDatabaseInstance_basic_with_user_labels|TestAccSqlDatabaseInstance_settings_upgrade|TestAccSqlDatabaseInstance_settings_deletionProtection|TestAccSqlDatabaseInstance_settings_secondary|TestAccSqlDatabaseInstance_settings_basic|TestAccSqlDatabaseInstance_settings_deletionProtectionEnabled|TestAccSqlDatabaseInstance_basicMSSQL|TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample|TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7713-old..auto-pr-7713) ( 2 files changed, 159 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7713-old..auto-pr-7713) ( 6 files changed, 1008 insertions(+), 2 deletions(-))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-7713-old..auto-pr-7713) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-7713-old..auto-pr-7713) ( 8 files changed, 219 insertions(+))
## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_network_security_address_group` (4 total tests)
Untested fields: `self_link`, `labels`

Please add acceptance tests which include these fields.

modular-magician commented:
#### Tests analytics 
Total tests: `2623` 
Passed tests `2324` 
Skipped tests: `277` 
Affected tests: `22` 

#### Action taken 
<details> <summary>Found 22 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccNetworkSecurityAddressGroup_networkSecurityAddressGroupsBasicExample|TestAccSqlDatabaseInstance_Timezone|TestAccSqlDatabaseInstance_rootPasswordShouldBeUpdatable|TestAccSqlDatabaseInstance_activationPolicy|TestAccSqlDatabaseInstance_updateReadReplicaWithBinaryLogEnabled|TestAccSqlDatabaseInstance_withPrivateNetwork_withoutAllocatedIpRange|TestAccSqlDatabaseInstance_authNets|TestAccSqlDatabaseInstance_PointInTimeRecoveryEnabled|TestAccSqlDatabaseInstance_PointInTimeRecoveryEnabledForSqlServer|TestAccSqlDatabaseInstance_basic_with_user_labels|TestAccNetworkSecurityAddressGroups_update|TestAccNetworkSecurityAddressGroup_networkSecurityAddressGroupsAdvancedExample|TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample|TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example|TestAccSqlDatabaseInstance_settingsDowngrade|TestAccSqlDatabaseInstance_settings_basic|TestAccSqlDatabaseInstance_settings_secondary|TestAccSqlDatabaseInstance_basicMSSQL|TestAccSqlDatabaseInstance_settings_upgrade|TestAccDataSourceGoogleFirebaseAndroidAppConfig|TestAccSqlDatabaseInstance_settings_deletionProtection|TestAccSqlDatabaseInstance_settings_deletionProtectionEnabled </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccNetworkSecurityAddressGroup_networkSecurityAddressGroupsBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccNetworkSecurityAddressGroup_networkSecurityAddressGroupsBasicExample.log)]
`TestAccSqlDatabaseInstance_rootPasswordShouldBeUpdatable`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_rootPasswordShouldBeUpdatable.log)]
`TestAccSqlDatabaseInstance_activationPolicy`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_activationPolicy.log)]
`TestAccSqlDatabaseInstance_updateReadReplicaWithBinaryLogEnabled`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_updateReadReplicaWithBinaryLogEnabled.log)]
`TestAccSqlDatabaseInstance_withPrivateNetwork_withoutAllocatedIpRange`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_withPrivateNetwork_withoutAllocatedIpRange.log)]
`TestAccSqlDatabaseInstance_authNets`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_authNets.log)]
`TestAccSqlDatabaseInstance_PointInTimeRecoveryEnabled`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_PointInTimeRecoveryEnabled.log)]
`TestAccSqlDatabaseInstance_PointInTimeRecoveryEnabledForSqlServer`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_PointInTimeRecoveryEnabledForSqlServer.log)]
`TestAccNetworkSecurityAddressGroups_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccNetworkSecurityAddressGroups_update.log)]
`TestAccNetworkSecurityAddressGroup_networkSecurityAddressGroupsAdvancedExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccNetworkSecurityAddressGroup_networkSecurityAddressGroupsAdvancedExample.log)]
`TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample.log)]
`TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example.log)]
`TestAccSqlDatabaseInstance_settingsDowngrade`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_settingsDowngrade.log)]
`TestAccSqlDatabaseInstance_settings_basic`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_settings_basic.log)]
`TestAccSqlDatabaseInstance_settings_secondary`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_settings_secondary.log)]
`TestAccSqlDatabaseInstance_settings_upgrade`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_settings_upgrade.log)]
`TestAccDataSourceGoogleFirebaseAndroidAppConfig`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccDataSourceGoogleFirebaseAndroidAppConfig.log)]
`TestAccSqlDatabaseInstance_settings_deletionProtection`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_settings_deletionProtection.log)]
`TestAccSqlDatabaseInstance_settings_deletionProtectionEnabled`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_settings_deletionProtectionEnabled.log)] 

Tests failed during RECORDING mode:
 `TestAccSqlDatabaseInstance_Timezone`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/build-log/recording_build/TestAccSqlDatabaseInstance_Timezone_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording/TestAccSqlDatabaseInstance_Timezone.log)] 

Several tests got terminated during RECORDING mode.
Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/8b9630c8-0dbb-412d-b0cd-95c7c05111cb/recording) for each test

SarahFrench commented:
Hi @diogoEsteves42 , sorry for not doing a proper review yet - I'm on bug hunting duty this week so I'm juggling a few things! I'll trigger the tests and give this PR a review once the tests complete 🚀

diogoEsteves42 commented:
Thanks for the message @SarahFrench, wish you can handle everything 💪🏼 

Ok about not merging the main.

SarahFrench commented:
There's been an issue with the tests running on this PR - I'll run /gcbrun just in case it helps

SarahFrench commented:
/gcbrun

SarahFrench commented:
There's a problem with the tests at the moment - please don't push any more commit! Thanks

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7713-old..auto-pr-7713) ( 2 files changed, 155 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7713-old..auto-pr-7713) ( 6 files changed, 978 insertions(+), 2 deletions(-))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-7713-old..auto-pr-7713) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-7713-old..auto-pr-7713) ( 8 files changed, 219 insertions(+))

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccSqlDatabaseInstance_rootPasswordShouldBeUpdatable`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/1b29584a-6377-4ebc-85c1-9eb5db2eff0d/recording/TestAccSqlDatabaseInstance_rootPasswordShouldBeUpdatable.log)]
`TestAccFirebaserulesRelease_BasicRelease`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/1b29584a-6377-4ebc-85c1-9eb5db2eff0d/recording/TestAccFirebaserulesRelease_BasicRelease.log)]
`TestAccNetworkSecurityAddressGroups_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/1b29584a-6377-4ebc-85c1-9eb5db2eff0d/recording/TestAccNetworkSecurityAddressGroups_update.log)]
`TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/1b29584a-6377-4ebc-85c1-9eb5db2eff0d/recording/TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample.log)]
`TestAccAlloydbBackup_missingLocation`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/1b29584a-6377-4ebc-85c1-9eb5db2eff0d/recording/TestAccAlloydbBackup_missingLocation.log)]
`TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/1b29584a-6377-4ebc-85c1-9eb5db2eff0d/recording/TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example.log)]
`TestAccAlloydbCluster_missingLocation`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/1b29584a-6377-4ebc-85c1-9eb5db2eff0d/recording/TestAccAlloydbCluster_missingLocation.log)]
`TestAccSqlDatabaseInstance_PointInTimeRecoveryEnabledForSqlServer`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/1b29584a-6377-4ebc-85c1-9eb5db2eff0d/recording/TestAccSqlDatabaseInstance_PointInTimeRecoveryEnabledForSqlServer.log)]
`TestAccDataSourceGoogleFirebaseAndroidAppConfig`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/1b29584a-6377-4ebc-85c1-9eb5db2eff0d/recording/TestAccDataSourceGoogleFirebaseAndroidAppConfig.log)] 

Tests failed during RECORDING mode:
 `TestAccTPUNode_tpuNodeFullExample`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/1b29584a-6377-4ebc-85c1-9eb5db2eff0d/build-log/recording_build/TestAccTPUNode_tpuNodeFullExample_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/1b29584a-6377-4ebc-85c1-9eb5db2eff0d/recording/TestAccTPUNode_tpuNodeFullExample.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/1b29584a-6377-4ebc-85c1-9eb5db2eff0d/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/1b29584a-6377-4ebc-85c1-9eb5db2eff0d/recording) for each test

SarahFrench commented on mmv1/products/networksecurity/AdressGroup.yaml:
```diff
@@ -0,0 +1,121 @@
+      Set of label tags associated with the AddressGroup resource.
+      An object containing a list of "key": value pairs. Example: { "name": "wrench", "mass": "1.3kg", "count": "3" }.
+  - !ruby/object:Api::Type::Enum
+    name: "type"
```
The API's docs say that this field is required - could you please add `required: true` in this section?

diogoEsteves42 replied:
Sure, thanks. Done

SarahFrench commented on mmv1/products/networksecurity/AdressGroup.yaml:
```diff
@@ -0,0 +1,121 @@
+  - !ruby/object:Api::Type::Array
+    name: "items"
+    item_type: Api::Type::String
+    required: true
```
From the API docs it seems this field is optional - could you please update this in the yaml?

https://cloud.google.com/traffic-director/docs/reference/network-security/rest/v1beta1/organizations.locations.addressGroups#AddressGroup.FIELDS.items

SarahFrench commented on mmv1/third_party/terraform/tests/resource_network_security_address_groups_test.go.erb:
```diff
@@ -0,0 +1,74 @@
+    labels      = {
+		foo = "foo"
+    }
+    items = ["208.80.154.224/32"]
```
Sometimes we find that list ordering can be an issue with providers - could you please add another item to this array and place them 'out of order', e.g. ["208.80.15**5**.224/32", "208.80.15**4**.224/32"]. Thanks!

diogoEsteves42 replied:
Ohh, I'll pay attention to that, but I didn't see anything related. Tested locally and works, thanks.

SarahFrench reviewed (commented):
Thanks for bearing with me and the test issues - here's a review 😅 . I found a few small things that need attention but otherwise everything looks good!

diogoEsteves42 commented:
Hi @SarahFrench, thank you for the review, about the field required and the other not, maybe I just switched them, appreciate the review.  About the test scenario I've added the item you just mentioned ran the test locally and it worked well.
Again, thanks for the review process.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7713-old..auto-pr-7713) ( 2 files changed, 155 insertions(+))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7713-old..auto-pr-7713) ( 6 files changed, 979 insertions(+), 2 deletions(-))
TF Conversion: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-7713-old..auto-pr-7713) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-7713-old..auto-pr-7713) ( 8 files changed, 219 insertions(+))

modular-magician commented:
#### Tests analytics 
Total tests: `2670` 
Passed tests `2382` 
Skipped tests: `280` 
Affected tests: `8` 

#### Action taken 
<details> <summary>Found 8 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests</summary><blockquote>TestAccFirebaserulesRelease_BasicRelease|TestAccNetworkSecurityAddressGroups_update|TestAccDataSourceGoogleFirebaseAndroidAppConfig|TestAccAlloydbBackup_missingLocation|TestAccAlloydbCluster_missingLocation|TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example|TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample|TestAccDataSourceAlloydbLocations_basic </blockquote></details> 

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccFirebaserulesRelease_BasicRelease`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/3febd67a-54c3-4afd-88f3-8af0abc87b87/recording/TestAccFirebaserulesRelease_BasicRelease.log)]
`TestAccNetworkSecurityAddressGroups_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/3febd67a-54c3-4afd-88f3-8af0abc87b87/recording/TestAccNetworkSecurityAddressGroups_update.log)]
`TestAccDataSourceGoogleFirebaseAndroidAppConfig`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/3febd67a-54c3-4afd-88f3-8af0abc87b87/recording/TestAccDataSourceGoogleFirebaseAndroidAppConfig.log)]
`TestAccAlloydbBackup_missingLocation`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/3febd67a-54c3-4afd-88f3-8af0abc87b87/recording/TestAccAlloydbBackup_missingLocation.log)]
`TestAccAlloydbCluster_missingLocation`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/3febd67a-54c3-4afd-88f3-8af0abc87b87/recording/TestAccAlloydbCluster_missingLocation.log)]
`TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/3febd67a-54c3-4afd-88f3-8af0abc87b87/recording/TestAccApigeeKeystoresAliasesPkcs12_ApigeeKeystoresAliasesPkcs12Example.log)]
`TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/3febd67a-54c3-4afd-88f3-8af0abc87b87/recording/TestAccApigeeKeystoresAliasesKeyCertFile_apigeeKeystoresAliasesKeyCertFileTestExample.log)]
`TestAccDataSourceAlloydbLocations_basic`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/3febd67a-54c3-4afd-88f3-8af0abc87b87/recording/TestAccDataSourceAlloydbLocations_basic.log)] 

All tests passed
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/3febd67a-54c3-4afd-88f3-8af0abc87b87/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7713/artifacts/3febd67a-54c3-4afd-88f3-8af0abc87b87/recording) for each test

diogoEsteves42 commented:
Hi @SarahFrench, everything ran with success, could we follow the review process, because I will need to update part of the example documentation of another resource with the usage of `addressGroup`.  Appreciate your attention
